python/caffe/_caffe.cpp

```diff
@@ -117,17 +117,17 @@ void Net_Save(const Net<Dtype>& net, string filename) {
 void Net_SetInputArrays(Net<Dtype>* net, bp::object data_obj,
     bp::object labels_obj, bp::object layer_name) {
 
-  const string str_layer = bp::extract<std::string>(layer_name);
-
   shared_ptr<MemoryDataLayer<Dtype> > md_layer;
-  if (str_layer == "") {
+  if (layer_name.ptr() == Py_None) {
```

Comment:
I will fix that with your solution, but to compare with Py_None is more clean, since we are dealing with the python bindings.
The method `.ptr()` will returns a `PyObject*` and the comparison with Py_None is the right way to do it.